src/app/researcher/study/request/[orgSlug]/actions.ts

```diff
@@ -156,3 +156,93 @@ export const onDeleteStudyAction = new Action('onDeleteStudyAction')
         // Clean up the files from s3
         await deleteFolderContents(`studies/${orgSlug}/${studyId}`)
     })
+
+const addJobToStudyActionArgsSchema = z.object({
+    studyId: z.string(),
+    orgSlug: z.string(),
+    mainCodeFileName: z.string(),
+    codeFileNames: z.array(z.string()),
+})
+
+export const addJobToStudyAction = new Action('addJobToStudyAction')
+    .params(addJobToStudyActionArgsSchema)
+    .middleware(async ({ params: { studyId } }) => {
+        const { orgId } = await getStudyOrgIdForStudyId(studyId)
+        return { orgId }
+    })
+    .requireAbilityTo('update', 'Study')
+    .handler(async ({ params: { studyId, orgSlug, mainCodeFileName, codeFileNames }, session }) => {
+        const userId = session.user.id
+
+        // Create new job for existing study
+        const studyJob = await db
+            .insertInto('studyJob')
+            .values({
+                language: 'R',
+                studyId: studyId,
+            })
+            .returning('id')
+            .executeTakeFirstOrThrow()
+
+        await db
+            .insertInto('jobStatusChange')
+            .values({
+                studyJobId: studyJob.id,
+                status: 'INITIATED',
+            })
+            .executeTakeFirstOrThrow()
+
+        await db
+            .insertInto('jobStatusChange')
+            .values({
+                userId: userId,
+                status: 'CODE-SUBMITTED',
+                studyJobId: studyJob.id,
+            })
+            .execute()
+
+        await db
+            .insertInto('studyJobFile')
+            .values({
+                name: mainCodeFileName,
+                path: pathForStudyJobCodeFile({ orgSlug, studyId, studyJobId: studyJob.id }, mainCodeFileName),
+                studyJobId: studyJob.id,
+                fileType: 'MAIN-CODE',
+            })
+            .executeTakeFirstOrThrow()
+
+        for (const fileName of codeFileNames) {
+            await db
+                .insertInto('studyJobFile')
+                .values({
+                    name: fileName,
+                    path: pathForStudyJobCodeFile({ orgSlug, studyId, studyJobId: studyJob.id }, fileName),
+                    studyJobId: studyJob.id,
+                    fileType: 'SUPPLEMENTAL-CODE',
+                })
+                .executeTakeFirstOrThrow()
+        }
```

Comment:
the above code is identical to most of `onCreateStudyAction`.  lets extract the common parts out to a 'addStudyJob` function that most actions call.  Could live in this file or maybe db/mutations